client/packages/programs/src/JsonForms/components/PreviousEncounterText.tsx

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import { ControlProps, rankWith, uiTypeIs } from '@jsonforms/core';
+import { withJsonFormsControlProps } from '@jsonforms/react';
+import { DetailInputWithLabelRow } from '@openmsupply-client/common';
+import { FORM_LABEL_WIDTH } from '../common';
+import { useEncounter } from '../../api';
+import { get as extractProperty } from 'lodash';
+
+export const previousEncounterTextTester = rankWith(
+  10,
+  uiTypeIs('PreviousEncounterText')
+);
+
+const UIComponent = (props: ControlProps) => {
+  const { label, path } = props;
+
+  // fetch current encounter
+  const encounterId = useEncounter.utils.idFromUrl();
+  const { data: currentEncounter } = useEncounter.document.byId(encounterId);
+
+  // fetch previous encounter
+  const { data: previousEncounter } = useEncounter.document.previous(
+    currentEncounter?.patient.id,
+    currentEncounter?.startDatetime
+      ? new Date(currentEncounter?.startDatetime)
+      : new Date()
+  );
+
+  if (!props.visible) {
+    return null;
+  }
```

Comment:
Because this is such a simple component display-wise, I wonder if we should make a more generic "TextLookup" component, which takes the database query as a parameter?

I guess defining what the query is in the ui schema might be a bit tricky, but something to think about next time we have to do a similar element.

Reply:
You mean just pass in the query hook as a parameter? Yes that could be useful! especially because I think we could need a "status" display control which is basically the same but uses a different query...

Reply:
> You mean just pass in the query hook as a parameter?

Yeah, that could be a way to do it. We should probably have a map somewhere that lists "common" names to use as parameters mapped to specific hooks/queries. So the uischema would just have something like:

```
"options": {
  "query": "previousRegimen"
}
```
and map that to the query you used here.

Something like that.